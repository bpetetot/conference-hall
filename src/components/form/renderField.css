.form-label {
  align-items: center;
  display: flex;
  width: 100%;
}

.form-label > label {
  color: var(--menu-color-header);
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
  min-width: 20%;
}

.form-label > div {
  width: 100%;
}

.form-has-error > label {
  color: var(--error-color);
}

.form-has-error > input {
  border: 1px solid var(--error-color);
}

.form-has-error > textarea {
  border: 1px solid var(--error-color);
}

@media screen and (max-width: 640px) {
  .form-label {
    display: block;
  }
  .form-label > label {
    min-width: 100%;
  }
}
