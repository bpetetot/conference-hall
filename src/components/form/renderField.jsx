/* eslint-disable react/prop-types */
import React from 'react'

import Toggle from './toggle'
import Address from './address'
import MarkdownInput from './markdownInput'
import { DayPicker, DayRangePicker } from './dayPicker'

import Label from './label'

const renderField = component => ({
  input, label, type, meta, placeholder, children, ...rest
}) => (
  <Label name={input.name} label={label} error={meta.error}>
    {component === 'address' && <Address {...input} autoComplete="nope" />}
    {component === 'input' && <input {...input} id={input.name} type={type} placeholder={placeholder} />}
    {component === 'select' && <select {...input} id={input.name}>{children}</select>}
    {component === 'textarea' && (
      <textarea id={input.name} {...input}>
        {input.value}
      </textarea>
    )}
    {component === 'markdown-input' && <MarkdownInput {...input} />}
    {component === 'dayPicker' && <DayPicker id={input.name} {...input} />}
    {component === 'dayRangePicker' && <DayRangePicker id={input.name} {...input} />}
    {component === 'toggle' && <Toggle {...rest} {...input} name={input.name} />}
  </Label>
)

export default renderField
