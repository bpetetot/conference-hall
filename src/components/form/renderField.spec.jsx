/* eslint-env jest */
import { shallow } from 'enzyme'
import React from 'react'
import renderField from './renderField'

describe('components/renderField', () => {
  it('should render', () => {
    const MyInput = renderField('input')
    const wrapper = shallow(<MyInput meta={{ error: 'error' }} input={{ name: 'boo' }} type="radio" placholder="placeholder" />)
    expect(wrapper).toMatchSnapshot()
  })
})
