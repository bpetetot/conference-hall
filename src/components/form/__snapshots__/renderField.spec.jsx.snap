// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/renderField should render 1`] = `
<WithSizes
  error="error"
  name="boo"
>
  <input
    id="boo"
    name="boo"
    type="radio"
  />
</WithSizes>
`;
